lib/gemstash/cli/preload.rb

```diff
@@ -0,0 +1,31 @@
+require "gemstash"
+
+module Gemstash
+  class CLI
+    # This implements the command line preload task to cache all the available gems:
+    # $ gemstash preload
+    class Preload
+      include Gemstash::Env::Helper
+
+      def initialize(cli)
+        Gemstash::Env.current = Gemstash::Env.new
+        @cli = cli
+      end
+
+      def run
+        return unless are_you_sure?
+        http_client = HTTPClient.for(Upstream.new(@cli.options[:server_url]))
+        preloader = Gemstash::Preload::GemPreloader.new(http_client, latest: @cli.options[:latest]).
+          limit(@cli.options[:limit]).skip(@cli.options[:skip]).threads(@cli.options[:threads])
```

Comment:
I meant the constructor of `Gemstash::Preload::GemPreloader`, so something like this:
```
preloader = Gemstash::Preload::GemPreloader.new(http_client, @cli.options)
```

And then the `Gemstash::Preload::GemPreloader` would store all the `:latest`, `:limit`, `:skip`, and `:threads` options directly rather than using additional builder methods (or perhaps store the options object directly and use them as needed from there).